lib/withOnyx/index.tsx

```diff
@@ -249,13 +271,13 @@ export default function (mapOnyxToState, shouldDelayUpdates = false) {
                 // We will add this key to our list of recently accessed keys
                 // if the canEvict function returns true. This is necessary criteria
                 // we MUST use to specify if a key can be removed or not.
-                _.each(mapOnyxToState, (mapping) => {
-                    if (_.isUndefined(mapping.canEvict)) {
+                mapOnyxToStateEntries(mapOnyxToState).forEach(([, mapping]) => {
+                    if (mapping.canEvict === undefined) {
                         return;
                     }
 
-                    const canEvict = Str.result(mapping.canEvict, this.props);
-                    const key = Str.result(mapping.key, this.props);
+                    const canEvict = Str.result(mapping.canEvict as GenericFunction, this.props) as boolean;
```

Comment:
It looks like it should work as well
```suggestion
                    const canEvict = !!Str.result(mapping.canEvict as GenericFunction, this.props);
```